### DISCUSSION TITLE
CPU intensive Background Type Checking

### DISCUSSION THREAD

**Discussion by ollipekka** [ionide/ionide-vscode-fsharp#1879] [Q&A]
Somewhere among the latest releases I have noticed that every time I save, Ionide launches very CPU intensive background type checking. I didn't have this before, so something must have changed. The consequence of this is that the fans of my computer blow constantly, which is why I started to notice this in the first place.

Is there any way to disable this, or partially disable this? I'd like to keep the type hints, but have the type checking run when I compile. My largest project has around 100 files and I have multiple projects.

**Comment by ollipekka** [comment]
Is "FSharp.enableBackgroundServices": false supposed to disable the type checking? I tried to add this to the user config, but the type checking still kicks off on save. The irony here is that the incremental compile takes much less than the type checking on my machine.

**Reply by vzarytovskii** [reply]
You can't just "disable" the typechecking, otherwise you won't get type hints, lens, diagnostics and autocompletions.

Some subset of your type graph will always have to be re-typechecked, every time you change and save something.
